src/super_gradients/training/utils/pose_estimation/dekr_decode_callbacks.py

```diff
@@ -0,0 +1,299 @@
+    :param keypoint_threshold: (float)
+    :return:
+    """
+    maxm = hierarchical_pool(heatmap)
+    maxm = torch.eq(maxm, heatmap).float()
+    heatmap = heatmap * maxm
+    scores = heatmap.view(-1)
+    scores, pos_ind = scores.topk(max_num_people)
+
+    select_ind = (scores > (keypoint_threshold)).nonzero()
+    scores = scores[select_ind][:, 0]
+    pos_ind = pos_ind[select_ind][:, 0]
+
+    return pos_ind, scores
+
+
+def up_interpolate(x, size):
+    H = x.size(2)
+    W = x.size(3)
+    scale_h = int(size[0] / H)
+    scale_w = int(size[1] / W)
+    inter_x = torch.nn.functional.interpolate(x, size=[size[0] - scale_h + 1, size[1] - scale_w + 1], align_corners=True, mode="bilinear")
+    padd = torch.nn.ReplicationPad2d((0, scale_w - 1, 0, scale_h - 1))
+    return padd(inter_x)
+
+
+def cal_area_2_torch(v):
+    w = torch.max(v[:, :, 0], -1)[0] - torch.min(v[:, :, 0], -1)[0]
+    h = torch.max(v[:, :, 1], -1)[0] - torch.min(v[:, :, 1], -1)[0]
+    return w * w + h * h
+
+
+def nms_core(pose_coord, heat_score, nms_threshold: float, nms_num_threshold: int):
+    num_people, num_joints, _ = pose_coord.shape
+    pose_area = cal_area_2_torch(pose_coord)[:, None].repeat(1, num_people * num_joints)
+    pose_area = pose_area.reshape(num_people, num_people, num_joints)
+
+    pose_diff = pose_coord[:, None, :, :] - pose_coord
+    pose_diff.pow_(2)
+    pose_dist = pose_diff.sum(3)
+    pose_dist.sqrt_()
+    pose_thre = nms_threshold * torch.sqrt(pose_area)
+    pose_dist = (pose_dist < pose_thre).sum(2)
+    nms_pose = pose_dist > nms_num_threshold
+
+    ignored_pose_inds = []
+    keep_pose_inds = []
+    for i in range(nms_pose.shape[0]):
+        if i in ignored_pose_inds:
+            continue
+        keep_inds = nms_pose[i].nonzero().cpu().numpy()
+        keep_inds = [list(kind)[0] for kind in keep_inds]
+        if len(keep_inds) == 0:
+            continue
+        keep_scores = heat_score[keep_inds]
+        ind = torch.argmax(keep_scores)
+        keep_ind = keep_inds[ind]
+        if keep_ind in ignored_pose_inds:
+            continue
+        keep_pose_inds += [keep_ind]
+        ignored_pose_inds += list(set(keep_inds) - set(ignored_pose_inds))
+
+    return keep_pose_inds
+
+
+def get_heat_value(pose_coord, heatmap):
+    _, h, w = heatmap.shape
+    heatmap_nocenter = heatmap[:-1].flatten(1, 2).transpose(0, 1)
+
+    y_b = torch.clamp(torch.floor(pose_coord[:, :, 1]), 0, h - 1).long()
+    x_l = torch.clamp(torch.floor(pose_coord[:, :, 0]), 0, w - 1).long()
+    heatval = torch.gather(heatmap_nocenter, 0, y_b * w + x_l).unsqueeze(-1)
+    return heatval
+
+
+def pose_nms(heatmap_avg, poses, max_num_people: int, nms_threshold: float, nms_num_threshold: int):
+    """
+    NMS for the regressed poses results.
+
+    Args:
+        heatmap_avg (Tensor): Avg of the heatmaps at all scales (1, 1+num_joints, w, h)
+        poses (List): Gather of the pose proposals [(num_people, num_joints, 3)]
```

Comment:
Minor note: it's not the same convention as usually (`:param heatmap_avg:`).
Same for `aggregate_results`.

Reply:
Good catch. Not sure why pycharm generated this type of docstrings